Artifacts/windows-vsts-download-and-run-script/DownloadVstsDropAndExecuteScript.ps1

```diff
@@ -49,16 +44,52 @@ function Handle-LastError
     exit -1
 }
 
-function Set-AuthHeaders
+###################################################################################################
+#
+# Functions used in this script.
+#
+
+function Get-BuildArtifacts
 {
     [CmdletBinding()]
     param (
-        [string] $UserName,
-        [string] $AccessToken
+        [string] $ArtifactsUri,
+        [Hashtable] $Headers,
+        [string] $Destination
     )
 
-    $basicAuth = [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes("$UserName`:$AccessToken"))
-    return @{ Authorization = "Basic $basicAuth" }
+    # Clean up destination path first, if needed.
+    if (Test-Path $Destination -PathType Container)
+    {
+        Write-Host "Cleaning up destination folder $Destination"
+        Remove-Item -Path $Destination -Force -Recurse | Out-Null
+    }
+
+    Write-Host "Getting build artifacts information from $ArtifactsUri"
+    [Array] $artifacts = (Invoke-RestMethod -Uri $ArtifactsUri -Headers $Headers -Method Get | ConvertTo-Json -Depth 3 | ConvertFrom-Json).value
+
+    # Process all artifacts found.
+    foreach ($artifact in $artifacts)
+    {
+        $artifactName = "$($artifact.name)"

```

Comment:
Couldn't "$($artifact.name)" be simplified to $artifact.name?

Reply:
Unfortunately, "$($artifact.name)" cannot be simplified to $artifact.name.  PowerShell supports expanding variables or expressions, but expressions (even ones that just access a property on a variable) must be enclosed in parenthesis.  The PowerShell team wrote an interesting blog explaining [variable expansion and Here-Strings](https://blogs.msdn.microsoft.com/powershell/2006/07/15/variable-expansion-in-strings-and-here-strings/).

Reply:
Like Elizabeth said. Additionally, because the artifact name can have spaces, we want to make sure they are maintained. Although, just putting `$artifact.name` should also work.